.travis.yml

```diff
@@ -1,27 +1,67 @@
 sudo: required
+
 language: scala
+
 jdk:
-  - oraclejdk8
+- oraclejdk8
+
 addons:
   apt:
-    packages: # Required for docker build
-      - net-tools
-# These directories are cached to S3 at the end of the build
+    packages:
+    - net-tools
+
 cache:
   directories:
-    - $HOME/.ivy2/cache
-    - $HOME/.sbt/boot/
+  - "$HOME/.ivy2/cache"
+  - "$HOME/.sbt/boot/"
+
 before_cache:
-  # Tricks to avoid unnecessary cache updates
   - find $HOME/.ivy2 -name "ivydata-*.properties" -delete
   - find $HOME/.sbt -name "*.lock" -delete
+
 services:
   - docker
+
 scala:
   - 2.11.8
+
 env:
   - DOCKER_DIR="../docker/"
+
 script:
   - sbt ci
+
 after_success:
   - bash <(curl -s https://codecov.io/bash)
+
+before_deploy:
+  - sbt assembly
+  - export JAR_DIR="telemetry-streaming"
+  - export JAR_NAME="telemetry-streaming.jar"
+  - mkdir $JAR_DIR
+  - mv target/scala-2.11/telemetry-streaming-assembly-0.1-SNAPSHOT.jar "$JAR_DIR/$JAR_NAME"
+
+deploy:
+  - provider: s3
+    bucket: telemetry-artifacts-stage
+    local_dir: $JAR_DIR
+    skip_cleanup: true
+    region: us-west-2
+    secret_access_key:
```

Comment:
Can these encrypted values be moved into environment variables?

Reply:
I was thinking about setting some env vars to the secret values above but I am not sure if that would even work. I think it's fine the way it is, I am just paranoid about having these values (even encrypted) available.

Did you provision a new user + api key for this? If not just let me know and I can help.